docs/dev/RampUpExample.md

```diff
@@ -0,0 +1,204 @@
+    ```
+    kubectl apply -f https://raw.githubusercontent.com/risingwavelabs/risingwave-operator/main/docs/manifests/risingwave/risingwave-in-memory.yaml
+    ```
+4. The operator is the plugin of the k8s. The k8s can only deal with part of the logic. For the logic cannot deal with, we should impelement them in controller. 
+
+# How to implement
+## 1. How to define the additional print column
+1. The `RUNNING`, `STORAGE(META)`, `STORAGE(OBJECT)`, `AGE` are [Additional Printer Columns](https://book.kubebuilder.io/reference/generating-crd.html#additional-printer-columns) defined by ourselves. 
+2. We get these value from yaml file then print.
+3. In the codebase, we define it [here](/apis/risingwave/v1alpha1/risingwave_types.go). We define them in comments in some specific format and the kubebuilder will parse it. We need to define `name`, `type`, and `JSONPath`. For detailed information about `JSONPath`, check [here](https://kubernetes.io/docs/reference/kubectl/jsonpath/).
+   
+```golang
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:shortName=rw,categories=all;streaming
+// +kubebuilder:printcolumn:name="RUNNING",type=string,JSONPath=`.status.conditions[?(@.type=="Running")].status`
+// +kubebuilder:printcolumn:name="STORAGE(META)",type=string,JSONPath=`.status.storages.meta.type`
+// +kubebuilder:printcolumn:name="STORAGE(OBJECT)",type=string,JSONPath=`.status.storages.object.type`
+// +kubebuilder:printcolumn:name="VERSION",type=string,JSONPath=`.status.version`
+// +kubebuilder:printcolumn:name="AGE",type=date,JSONPath=`.metadata.creationTimestamp`
+```
+
+
+## 2. Define the struct if needed
+
+We can use `JSONPath = .spec.XXX`, `JSONPath = .metadata.XXX` or `JSONPath = .status.XXX`. If you want to add some new field, in codebase, we need to change the struct field [here](../../apis/risingwave/v1alpha1/risingwave_types.go)
+
+```golang
+type RisingWave struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    Spec   RisingWaveSpec   `json:"spec,omitempty"`
+    Status RisingWaveStatus `json:"status,omitempty"`
+}
+```
+
+
+## 3. How to register CRD into k8s
+
+We use `make manifests` and `make install-local` to create a new YAML file. The command will generate new [Yaml file](/config/crd/bases/risingwave.risingwavelabs.com_risingwaves.yaml) according to [golang file](../../apis/risingwave/v1alpha1/risingwave_types.go), and regist the CRD `risingwave` into k8s. Then, the user can create the `risingwave` instance in k8s.
+
+
+## 4. How to update the column
+1. Firstly, because `JSONPath` can only support some simple logic, but sometime we want some complex logic. In order to achieve the complex logic, we need to define the field as `.status.XX`. 
+2. If you use the `metaData` or `Spec`, the value should be immutable, so we will not change them. But if you define the `status`, we need some extract work to update it. 
```

Comment:
"Immutable" doesn't sound like a precise word. Fields in the `metadata` and `spec` can be updated either by users or the controllers.

```suggestion
2. If you use the `metadata` or `spec`, the value should be immutable, so we will not change them. But if you define the `status`, we need some extra work to update it. 
```